src/write.h

```diff
@@ -30,18 +30,22 @@
 /**
  * Write to a data object from a local file using the put protocol.
  *
- * @param[in]  conn        An open iRODS connection.
- * @param[in]  obj_file    A local file name.
- * @param[in]  rods_path   An iRODS data object path.
- * @param[in]  flags       CALCULATE_CHECKSUM to calculate a checksum on
-                           the server side. WRITE_LOCK to use an advisory
-                           lock server-side. Optional.
- * @param[out] error       An error report struct.
+ * @param[in]  conn             An open iRODS connection.
+ * @param[in]  obj_file         A local file name.
+ * @param[in]  rods_path        An iRODS data object path.
+ * @param[in]  default_resource An iRODS resource name. Optional, may be NULL.
+ * @param[in]  flags            CALCULATE_CHECKSUM to calculate and register a
+                                checksum on the server side, VERIFY_CHECKSUM to
+                                calculate and register a checksum on the server
+                                side and verify against a client side checksum,
+                                WRITE_LOCK to use an advisory lock on the
+                                server side. Optional.
+ * @param[out] error            An error report struct.
  *
  * @return The number of bytes copied in total.
  */
 int put_data_obj(rcComm_t *conn, const char *local_path, rodsPath_t *rods_path,
-                 int flags, baton_error_t *error);
+                 char *resource, int flags, baton_error_t *error);
```

Comment:
The argument names are inconsistent between the comment and declaration (obj_file vs local_path, default_resource vs resource), and also between the declaration and definition which is on lines 28/29 of write.c (the same pair of arguments).

Reply:
Thanks. I've made those consistent.